experimental/rocm/tracing.c

```diff
@@ -10,11 +10,24 @@
 
 #include "experimental/rocm/status_util.h"
 
+#include <dlfcn.h>
+
 // Total number of events per tracing context. This translates to the maximum
 // number of outstanding timestamp queries before collection is required.
 // To prevent spilling pages we leave some room for the context structure.
 #define IREE_HAL_ROCM_TRACING_DEFAULT_QUERY_CAPACITY (16 * 1024 - 256)
 
+typedef void (*SmuDumpCallback)(uint64_t, const char*, const char*, double);
+typedef bool (*SmuDumpInitFunc) (SmuDumpCallback callback);
+typedef void (*SmuDumpEndFunc) (void);
+typedef void (*SmuDumpOnceFunc) (void);
+typedef void (*RegDumpOnceFunc) (void);
+typedef void (*SviDumpOnceFunc) (void);
+typedef uint32_t (*RegGetTraceRate) (void);
+typedef uint32_t (*SmuGetTraceRate) (void);
+
+#define SMUTRACE_DLL "libsmutrace.so"
```

Comment:
define this per platform as with other users of iree_dynamic_library_load_from_files